### DISCUSSION TITLE
How to use `{url}` and current directory (dirname) in `url.searchengines`

### DISCUSSION THREAD

**Discussion by tkna91** [qutebrowser/qutebrowser#7244] [Q&A] [answered]
Thanks for your cool project.

I would like to search for keywords under the directory where the current page is located as `url.searchengines` with the following image.

```
"gp": "https://www.google.co.jp/search?&q=site:{url} {}"
```
There are two problems here for me as follows.
Sorry if they are basic. If you know how to solve them, please let me know.

1. can't use `{url}` in `url.searchengines`
2. I don't know how to use the expression to specify the current directory (dirname) in the command.

**Comment by toofar** [comment]
1.

Is this meant to be used like you are currently on the site https://example.com/some/page and you want to search for instances of "foo" on site:example.com? If so you can't do that straight from the setting. url.searchengines doesn't support placeholders based on the current page.
You could write a  [userscript](https://qutebrowser.org/doc/userscripts.html) to do so though. That would take the search term and use the environment variables to construct the search URL then write it to the QUTE_FIFO.

2.

I'm not sure what you mean by that. Can you provide an example?

**Reply by The-Compiler** [reply]
as for 1., an alias mapping to `:open` and using `{url}` would probably suffice too.

**Reply by toofar** [reply]
Ah, so it does. Maybe {url:host} for the substitution though if I remember that feature correctly.

**Comment by tkna91** [comment]
@toofar @The-Compiler Thank you kindly.

I see 5 when I run 4 on 3 below, but was hoping for an easy way to change it to 6.

3. current page
https://qutebrowser.org/doc/userscripts.html

4. execution details
`:open -t https://www.google.com/search?q=site:{url} key`

5. search result of the relevant page
https://www.google.com/search?q=site%3Ahttps%3A//qutebrowser.org/doc/userscripts.html%20key

6. search result of relevant directory
https://www.google.com/search?q=site:https://qutebrowser.org/doc/%20key

Thanks to your guidance, I was able to create the following script to display 6.

7. `~/.local/share/qutebrowser/userscripts/dir`
```
#! /bin/bash
KEY=$*
DIR=$(dirname $QUTE_URL)
echo "open -t https://www.google.com/search?q=site:$DIR $KEY" >> "$QUTE_FIFO"
```

The following settings in `config.py` have created the key bindings, but I am unable to specify the search keywords, as in `{}` in the `url.searchengines` command.

8. key bind configuration
`config.bind(",d", 'spawn --userscript dir')`

Sorry for my poor reading comprehension of the manual, but I would appreciate it if you could help me.

**Reply by mschilli87** [reply]
@tkna91:
> https://www.google.com/search?q=site:https://qutebrowser.org/doc/help/%20key

I don't get where the `help/` part is coming from.

**Reply by tkna91** [reply]
Sorry. It's a mistake. I fixed it.

**Reply by The-Compiler** [reply]
I don't understand what you're trying to do with `DIR=$(dirname $QUTE_URL)` here. That would just give you `https:`. As outlined in my earlier reply, I don't think you will need a userscript: Does `:open -t https://www.google.com/search?q=site:{url:host} key` do what you want?

If yes, all you need is an alias or a keybinding to fill that into the commandline - for example, via `:bind ,s set-cmd-text -s :open -t https://www.google.com/search?q=site:{{url:host}}`.

**Reply by tkna91** [reply]
I wanted to search the parent directory of the current URL.
Intended to be used when the hostname is too broad for the search, but the current page is too narrow.

**Comment by tkna91** [comment, marked as answer]
Thanks to what you taught me, I was able to do what I wanted to do.

`config.py`
```
# Search by this hostname
config.bind(',h', "set-cmd-text -s :open -t https://www.google.com/search?q=site:{{url:host}}")

# Search by parent directory
config.bind(',p', "set-cmd-text -s :spawn --userscript search_parent_directory")

# Search by this URL
config.bind(',s', "set-cmd-text -s :open -t https://www.google.com/search?q=site:{{url}}")
```

`~/.local/share/qutebrowser/userscripts/search_parent_directory`
```
#!/bin/bash
KEY=$*
DIR=$(dirname $QUTE_URL)/
echo "open -t https://www.google.com/search?q=site:$DIR $KEY" >> "$QUTE_FIFO"
```
